plugins/modules/ndo_synce_interface_policy.py

```diff
@@ -0,0 +1,277 @@
+        host: mso_host
+        username: admin
+        password: SomeSecretPassword
+        template: ansible_fabric_policy_template
+        state: query
+    register: query_all
+
+- name: Delete a syncE interface policy
+    cisco.mso.ndo_synce_interface_policy:
+        host: mso_host
+        username: admin
+        password: SomeSecretPassword
+        template: ansible_fabric_policy_template
+        interface_policy: ansible_test_interface_policy
+        state: absent
+"""
+
+RETURN = r"""
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.template import MSOTemplate, KVPair
+from ansible_collections.cisco.mso.plugins.module_utils.constants import ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        dict(
+            template=dict(type="str", required=True),
+            interface_policy=dict(type="str", aliases=["name"]),
+            interface_policy_uuid=dict(type="str", aliases=["uuid"]),
+            description=dict(type="str"),
+            admin_state=dict(type="str", choices=["enabled", "disabled"], default="disabled"),
+            sync_state_msg=dict(type="str", choices=["enabled", "disabled"], default="disabled"),
+            selection_input=dict(type="str", choices=["enabled", "disabled"], default="disabled"),
+            src_priority=dict(type="int", default=100),
+            wait_to_restore=dict(type="int", default=5),
+            state=dict(type="str", choices=["absent", "query", "present"], default="query"),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["interface_policy"]],
+            ["state", "absent", ["interface_policy"]],
+        ]
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    interface_policy = module.params.get("interface_policy")
+    interface_policy_uuid = module.params.get("interface_policy_uuid")
+    description = module.params.get("description")
+    admin_state = module.params.get("admin_state")
+    sync_state_msg = module.params.get("sync_state_msg")
+    selection_input = module.params.get("selection_input")
+    src_priority = module.params.get("src_priority")
+    wait_to_restore = module.params.get("wait_to_restore")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "fabric_policy", template)
+    mso_template.validate_template("fabricPolicy")
+
+    path = "/fabricPolicyTemplate/template/syncEthIntfPolicies"
+
+    existing_interface_policies = mso_template.template.get("fabricPolicyTemplate", {}).get("template", {}).get("syncEthIntfPolicies", [])
+
+    if interface_policy:
```

Comment:
Can the object be updated with UUID? Please add a test for the update with UUID.

Reply:
yes, added test case